File: sn/src/main/java/org/sqlbuilder/sn/SnSensekeyResolver.java
package org.sqlbuilder.sn;

import org.sqlbuilder.common.Nullable;
import org.sqlbuilder.common.Resolver;
import org.sqlbuilder2.ser.DeSerialize;

import java.io.File;
import java.io.IOException;
import java.util.AbstractMap.SimpleEntry;

public class SnSensekeyResolver extends Resolver<String, SimpleEntry<Integer, Integer>>
{
	public SnSensekeyResolver(final String ser) throws IOException, ClassNotFoundException
	{
		super(DeSerialize.deserialize(new File(ser)));
	}
}

File: sn/src/main/java/org/sqlbuilder/sn/SnSensekeyResolver.kt
package org.sqlbuilder.sn

import org.sqlbuilder.common.DeSerialize
import org.sqlbuilder.common.Resolver
import java.io.File

typealias SnWordResolved = Int
typealias SnSynsetResolved = Int
typealias SnSensekeyResolvable = String
typealias SnSensekeyResolved = Pair<SnWordResolved, SnSynsetResolved>

class SnSensekeyResolver(ser: String) : Resolver<SnSensekeyResolvable, SnSensekeyResolved>(DeSerialize.deserialize<Map<SnSensekeyResolvable, SnSensekeyResolved>>(File(ser)))
